test/function/samples/transparent-dynamic-inlining/foo.js

```diff
@@ -0,0 +1,4 @@
+export var x = 42;
+import( './main' ).then(x => {
```

Comment:
As the console.log is not checked in the test, maybe we should instead export this promise and check it is a promise that resolves correctly? This would of course require the test code to be changed to wait if exports returns a promise but I do not think this would be a lot of work.

Reply:
I've adjusted these tests to actually run in the other cases now!

Reply:
Just checked, really nice!